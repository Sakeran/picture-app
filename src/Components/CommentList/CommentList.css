.CommentList {
    width: 90%;
    margin-top: 1em;
}

.CommentList h3 {
  margin-bottom: 2em;
}

.CommentList-List {
  list-style: none;
  padding: 0;
  margin-top: 1em;
}

.CommentList-no-comments h3 {
  margin: 1em 0;
  text-align: center;
}

.CommentList-load-btn {
  display: block;
  cursor: pointer;
  margin: 0 auto;
  margin-bottom: 0.5em;
  padding-bottom: 0.1em;
  border: none;
  background: none;
  color: #6d1cba;
  border-bottom: 1px solid #6d1cba;
}
