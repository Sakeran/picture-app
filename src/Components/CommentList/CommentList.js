import React from 'react';
import PropTypes from 'prop-types';

const CommentList = ({ comments, count, loadMore }) => (
  <div className="CommentList">
    <ul>
    {comments.map(e => (
      <li key={e.id}>
        {e.text}
      </li>
    ))}
    </ul>
    {
      (count && loadMore && (count > comments.length)) &&
      <button onClick={loadMore}>
        Load More Comments
      </button>
    }
  </div>
);

CommentList.propTypes = {
  comments: PropTypes.array.isRequired,
  count: PropTypes.number,
  loadMore: PropTypes.func
}

export default CommentList;
