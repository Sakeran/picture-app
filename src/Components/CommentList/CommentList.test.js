import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import CommentList from './CommentList';

const makeComment = (id) => ({
  id,
  text: 'Test Comment ' + id,
  user: {
    username: 'Test User'
  },
  date: 'January 1st 2001'
});

test('CommentList component renders correctly with comments', () => {
  const comments = [1,2,3].map(id => makeComment(id) );
  const props = {
    comments,
    count: 3,
    loadMore: jest.fn()
  };
  const component = shallow(<CommentList {...props} />);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
});

test('CommentList component renders correctly with no comments', () => {
  const props = {
    comments: [],
    count: 0,
    loadMore: jest.fn()
  };
  const component = shallow(<CommentList {...props} />);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
});
