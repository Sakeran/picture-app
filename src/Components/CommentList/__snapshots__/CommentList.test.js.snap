// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentList component renders correctly with comments 1`] = `
<div
  className="CommentList"
>
  <ul
    className="CommentList-List"
  >
    <li>
      <Comment
        comment={
          Object {
            "date": "January 1st 2001",
            "id": 1,
            "text": "Test Comment 1",
            "user": Object {
              "username": "Test User",
            },
          }
        }
      />
    </li>
    <li>
      <Comment
        comment={
          Object {
            "date": "January 1st 2001",
            "id": 2,
            "text": "Test Comment 2",
            "user": Object {
              "username": "Test User",
            },
          }
        }
      />
    </li>
    <li>
      <Comment
        comment={
          Object {
            "date": "January 1st 2001",
            "id": 3,
            "text": "Test Comment 3",
            "user": Object {
              "username": "Test User",
            },
          }
        }
      />
    </li>
  </ul>
</div>
`;

exports[`CommentList component renders correctly with no comments 1`] = `
<div
  className="CommentList CommentList-no-comments"
>
  <h3
    className="centered"
  >
    There are no comments for this post.
  </h3>
</div>
`;
